### PR TITLE
loglist: prefix match odd number of hex digits

### PR DESCRIPTION
**Pull request by daviddrysdale**

### CODE REVIEW

File: loglist/loglist.go [hunk 1]
```diff
@@ -168,6 +167,8 @@ func (ll *LogList) FindLogByKey(key []byte) *Log {
 	return nil
 }
 
+var hexDigits = regexp.MustCompile("^[0-9a-fA-F]+$")
```
Comment (Martin2112) [inline loglist/loglist.go]: can we use hex.DecodeString instead of a regex?

File: loglist/loglist.go [hunk 2]
```diff
@@ -206,7 +207,10 @@ func (ll *LogList) FuzzyFindLog(input string) []*Log {
 		if log := ll.FindLogByKey(data); log != nil {
 			return []*Log{log}
 		}
-		if logs := ll.FindLogByKeyHashPrefix(data); len(logs) > 0 {
+	}
+	// Finally, allow hex strings with an odd number of digits.
```
Comment (Martin2112) [inline loglist/loglist.go]: Needs a test?

Comment (daviddrysdale) [reply loglist/loglist.go]: Yep, done.

File: loglist/loglist.go (hunk 1 above)
Comment (daviddrysdale) [reply loglist/loglist.go]: `hex.DecodeString` doesn't allow odd number of digits.

Comment (Martin2112) [reply loglist/loglist.go]: How odd.

**Pull request merged**